presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -194,10 +199,11 @@ public synchronized void addLocation(URI location, TaskId remoteSourceTaskId)
         checkState(!noMoreLocations, "No more locations already set");
 
         RpcShuffleClient resultClient;
+        Optional<URI> asyncPageTransportLocation = getAsyncPageTransportLocaton(location, asyncPageTransportEnabled);
```

Comment:
Since it is only used here - you can either inline this method here, or move it into this class as a private static method.